src/ergw_tei_mngr.erl

```diff
@@ -128,16 +111,16 @@ alloc_tei_reply(From, Reply, Name, RndState, State, Data) ->
 
 alloc_tei(From, Name, _KeyFun, 0, RndState, State, Data) ->
     alloc_tei_reply(From, {error, no_tei}, Name, RndState, State, Data);
-alloc_tei(From, Name, KeyFun, Cnt, RndState0, State, #data{tid = TID} = Data0) ->
+alloc_tei(From, Name, KeyFun, Cnt, RndState0, State, Data0) ->
     Data = #data{prefix = Prefix0, prefix_len = PrefixLen} = maybe_update_data(Data0),
     Prefix = Prefix0 bsl (32 - PrefixLen),
     Mask = 16#ffffffff bsr PrefixLen,
     {TEI0, RndState} = rand:uniform_s(16#fffffffe, RndState0),
     TEI = Prefix + (TEI0 band Mask),
-    case ets:insert_new(TID, {KeyFun(TEI), Name}) of
-	true ->
+    case gtp_context_reg:lookup(KeyFun(TEI)) of
+	undefined ->
 	    alloc_tei_reply(From, {ok, TEI}, Name, RndState, State, Data);
```

Comment:
Do we need here add `ets:insert(gtp_context_reg, {KeyFun(TEI), Name})` as was implemented before in `gtp_context_reg.erl`?

Reply:
no, there context will register itself and thereby insert the teid. The is a small window in this we could create colliding TEIDs, but that was there before the change to teid_mngr.